packages/cli/generation/ir-generator/src/utils/getGenericDetails.ts

```diff
@@ -0,0 +1,15 @@
+export function getGenericDetails(name: string): { isGeneric: true; name?: string; arguments?: string[] } | undefined {
+    const genericMatch = name.match(/([\w.]+)<([\w,\s]+)>/);
+
+    if (genericMatch?.[0] != null) {
+        if (!genericMatch[1] || !new Set(["optional", "set", "list", "map", "literal"]).has(genericMatch[1]?.trim())) {
```

Comment:
where else are these hardcoded? i think this is going to get out of date too fast

Reply:
Only in these utility files now, of which there are 2.

Reply:
Is there a place for shared Regex?